### DISCUSSION TITLE
Enhancement: Include MariaDB as an App.

### DISCUSSION THREAD

**Discussion by Bira-oliverio** [IceWhaleTech/CasaOS#246] [Ideas]
A database is necessary for NextCloud app, please include MariaDB in the app store.

**System infomation**

	 - OS: linux
	 - CasaOS Version: 0.3.1.1
	 - Disk Total: 937291 
	 - Disk Used: 56583 
	 - Sync State: Sync img: true
	linuxserver/syncthing:latest,
	Sync Key length: 0
	 - System Info: [Ubuntu 22.04 LTS   \l Bit:64 Mem:15908M Core:8 Linux ubiratan-Macmini 5.15.0-33-generic #34-Ubuntu SMP Wed May 18 13:34:26 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux]
	 - Browser: Chrome 
	 - Version: 101.0.4951.61

**Comment by Protektor-Desura** [comment]
The docker setup/yml file should include the database server when it creates/sets up NextCloud. But a SQL database is not technically required depending on how you set things up. Like this example.

```
version: "3.3"
services:
  nextcloud:
    container_name: nextcloud
    environment:
      - PGID=1000
      - PUID=1000
      - TZ=America/Chicago
    image: "ghcr.io/linuxserver/nextcloud:latest"
    hostname: nextcloud
    networks:
      - archon
    ports:
      - "443:443/tcp"
    restart: unless-stopped
    volumes:
      - "/home/docker/nextcloud:/config:rw"
      - "/home/docker/nextcloud/data:/data:rw"


networks:
  archon:
    external: true```

Here is a link to another option.

https://github.com/nextcloud/all-in-one#how-to-use-this

**Comment by Bira-oliverio** [comment]
> 

Yes, we can use SQLite, but this is a very limited option.

**Comment by JohnGuan** [comment]
We currently use issues to track bugs, and I'm converting this issue into a discussion.

**Comment by Protektor-Desura** [comment]
You may instead want to use the NextCloud All-In-One docker version.
Included are:

Nextcloud
Nextcloud Office
High performance backend for Nextcloud Files
High performance backend for Nextcloud Talk
Backup solution (based on [BorgBackup](https://github.com/borgbackup/borg#what-is-borgbackup))
OnlyOffice
ClamAV

https://github.com/nextcloud/all-in-one

**Comment by tigerinus** [comment]
This post is not forgotten.

Soon CasaOS App will be docker compose based. By then, apps that requires multiple containers, e.g. Wordpress + MariaDB, will be supported.

Stay tuned.

**Reply by kmgolf08** [reply]
Any update on this?  Just got a Zima and trying to figure out how to install passbolt.  This seems to be needed in order to do that.

**Reply by tigerinus** [reply]
@kmgolf08 - native Docker Compose support has been added to v0.4.4-alpha.

Someone just needs to submit MariaDB as an app to the AppStore (it is community driven).

**Reply by WisdomSky** [reply]
> Any update on this? Just got a Zima and trying to figure out how to install passbolt. This seems to be needed in order to do that.

If you still need mariadb specially when you want it as your preferred db for nextcloud, there's a third-party appstore that contains mariadb. but you'd need to upgrade to v0.4.4 in order to use it.

https://github.com/WisdomSky/CasaOS-LinuxServer-AppStore

After installing mariadb, you need to open the mariadb app settings and update the user, pass and databasename environment variables then save.

**Reply by defencedog** [reply]
https://github.com/WisdomSky/CasaOS-LinuxServer-AppStore/blob/main/Apps/Mariadb/docker-compose.yml
https://github.com/WisdomSky/CasaOS-LinuxServer-AppStore/blob/main/Apps/Nextcloud/docker-compose.yml

Can we combine the x2 `.yml` into a single consolidated `.yml`?

**Reply by WisdomSky** [reply]
> https://github.com/WisdomSky/CasaOS-LinuxServer-AppStore/blob/main/Apps/Mariadb/docker-compose.yml https://github.com/WisdomSky/CasaOS-LinuxServer-AppStore/blob/main/Apps/Nextcloud/docker-compose.yml
> 
> Can we combine the x2 `.yml` into a single consolidated `.yml`?

yes you can.

**Reply by defencedog** [reply]
Can you please provide some instructions?

**Comment by 1337Reaper** [comment]
+1 for a database image to be included in the official "App Store". As this is marketed as a "server OS" a database application is absolutely needed.

**Comment by WisdomSky** [comment]
https://github.com/IceWhaleTech/CasaOS-AppStore/pull/179
